pallets/msa/src/lib.rs

```diff
@@ -614,11 +637,20 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// Add a provider to a delegator with the default permissions
-	pub fn add_provider(provider: Provider, delegator: Delegator) -> DispatchResult {
+	pub fn add_provider(
+		provider: Provider,
+		delegator: Delegator,
+		schemas: Vec<SchemaId>,
+	) -> DispatchResult {
 		ProviderInfoOf::<T>::try_mutate(delegator, provider, |maybe_info| -> DispatchResult {
 			ensure!(maybe_info.take() == None, Error::<T>::DuplicateProvider);
-
-			let info = ProviderInfo { permission: Default::default(), expired: Default::default() };
+			let granted_schemas = BoundedVec::<SchemaId, T::MaxSchemaGrants>::try_from(schemas)
+				.map_err(|_| Error::<T>::ExceedsMaxSchemaGrants)?;
+			let info = ProviderInfo {
+				permission: Default::default(),
+				expired: Default::default(),
+				schemas: OrderedSetExt::<SchemaId, T::MaxSchemaGrants>::from(granted_schemas),
+			};
 
```

Comment:
Check if at the time of adding this we arent exceeding maximum allowed schema grants and if not add that to provider info

Separate stories exists to a full CRUD on schema grants, this is just addition at time of registration

Reply:
Is there any possibility that we might need to check here that what if the `maybe_info` already exists? or that is not possible?

Reply:
I think at this stage we are simply adding a new one 
Question is , if a provider adds the same payload again do we overwrite?

Now when it comes to adding additional schemas, we will have another extrinsic tracked in a different story to do Update and Delete

Reply:
On another note, we are checking ensure!(maybe_info.take() == None, Error::<T>::DuplicateProvider);

Reply:
So i think that covers it and if a provider want to authorize additional schemas, they will use a different extrinsic